src/app/business/delfin/storages/storages.component.ts

```diff
@@ -0,0 +1,601 @@
+                });
+                
+                let capData = {
+                    labels: ['Used','Free'],
+                    datasets: [
+                        {
+                            data: [element['used_capacity'], element['free_capacity']],
+                            backgroundColor: [
+                                "#FF6384",
+                                "#45e800"
+                            ],
+                            hoverBackgroundColor: [
+                                "#FF6384",
+                                "#45e800"
+                            ]
+                        }]    
+                };
+
+                let opt = {
+                    legend:{
+                        position: 'right'
+                    }
+                }
+                element['capacityData'] = capData;
+                element['chartOptions'] = opt;
+                element['volumes'] = vols;
+                element['storagePools'] = pools;
+            });
+
+            // Invoke prepareTree() to prepare the tree structure for the widget
+            this.prepareTree(this.allStorages);
+            this.loading = false;
+
+        }, (error)=>{
+            console.log("Something went wrong. Could not fetch all storages", error);
+            this.loading = false;
+        })
+        
+    }
+
+    showChart(){
+        
+    }
+
+    getStorageArrayRawCapacity(storages){
+
+    }
+
+    getStorageArrayUsableCapacity(){
+
+    }
+
+    syncAllStorages(){
+
+    }
+
+    syncStorage(storageId){
+        this.ds.syncStorageById(storageId).subscribe((res)=>{
+            this.msgs = [];
+            this.msgs.push({severity: 'success', summary: 'Success', detail: 'Sync request has been sent. Please check back in some time.'});
+            this.getAllStorages();
+        }, (error)=>{
+            this.msgs = [];
+            let errorMsg = 'Error Syncing device.' + error.error_msg;
+            this.msgs.push({severity: 'error', summary: 'Error', detail: error});
+            console.log("Something went wrong. Could not initiate the sync.", error);
+        });
+    }
+    getVolumesByStorage(storageId){
+        this.ds.getAllVolumes().subscribe((res)=>{
+            this.allVolumes = res.json().volumes;
+            this.allVolumes.forEach(element => {
+                if(element['storage_id'] == storageId){
+                    this.volumesArr.push(element);
+                }
+            });
+            this.totalRecords = this.dataSource.length;
+            this.volumesArr = this.dataSource.slice(0, 10);
+            
+            return this.volumesArr;
+        }, (error)=>{
+            console.log("Something went wrong. Could not fetch Volumes.", error)
+            this.volumesArr = [];
+            return this.volumesArr;
+        });
+    }
+
+    getPoolsByStorage(storageId){
+        this.ds.getAllStoragePools().subscribe((res)=>{
```

Comment:
same here. Like volumes

Reply:
Got it. Will fix this in the subsequent PRs.

Reply:
If you are fixing in subsequent PR, please open issue for this

Reply:
Raised an issue to track this. #397